### ISSUE TITLE
Set default browser not working

### ISSUE THREAD

**Issue by midzer** [Alex313031/Thorium#5]
Hi again,

you can set the default browser in Chrom* via settings and inital toolbar button.

Somehow, this does not work for me in Thorium. Default browser is definetly not set.

Do you know what this might cause?

Cheers
midzer

**Comment by Alex313031** [comment]
@midzer This is an upstream bug. I noted it in https://github.com/Alex313031/Thorium/blob/main/BUGS.md I have pushed a "fix" that simply appends --no-default-browser-check in the .desktop file to prevent the annoying infobar every startup. You can still set it as the default in your window manager. In Ubunutu/PopOS/Gnome you can do that in settings. In xfce it's default applications. In openbox (what I use), I recommend galternatives.

What OS/DE are you using, and how did you find Thorium?

Thank you muchly for being my third official user, it keeps me motivated to keep improving Thorium and not just rebasing existing patches on chromium trunk. I just added a patch that will (supposed to) enable VAAPI on older intel chips, namely nahalem/sandy bridge/ivy bridge.
Also, the next release is changing the profile dir from $HOME/.config/chromium to $HOME/.config/thorium. This will allow running chromium and thorium side by side and will prevent user profile errors from version differences for people migrating from chromium to thorium.

If upgrading from an earlier thorium release (once this patch lands tommorow), I recommend sudo apt purge thorium-browser-unstable, then making the thorium dir in .config and copying all content from chromium into the thorium dir, then reinstalling. This will migrate your user profile dir over.

**Comment by midzer** [comment]
I am using Debian bullseye with Cinnamon Desktop and found Thorium via https://chromium.woolyss.com/#linux-64-bit and tested it immediately:

My Speedometer score went up 5 points (from 99 to 104) compared to latest Chrome Dev build (https://twitter.com/midzer/status/1467595495140900876). In addition Thorium "feels" way more fast and instant. Browsing with it is a real joy!

If I can support this project give me a hint, whether it's testing or else.

**Comment by Alex313031** [comment]
@midzer I'm using CrunchBangPlusPlus, a minimalist Openbox distro based on Debian Bullseye https://crunchbangplusplus.org/. And yeah, besides the various patches and installer changes, It is compiled with AVX, FMA, LOOP Optimizations, PGO, ThinLTO, and has CFlags and LDFlags set to 3, all of which increases the binary size by about 50mb compared to a vanilla chromium build, but increases performance substantially. Have you read the README.md? And yeah I've been using chromium.woolyss.com for forever, using the windows xp builds back in 2015 and still do (Im also a legacy OS enthusiast, ask about my IBM PS/2 machine and various win 95/98/NT/pentium iii/pentium4/core 2 duo machines lolol), using it on my win 7,10, and 11 drives, and using it on my linux drives until about 6 months ago when I started building chromium myself. I then got into contact with RobRich (we have a discussion here https://github.com/RobRich999/Chromium_Clang/issues/26  where we help each other improve chromium). Rob makes the releases right above mine on Chromium.wooylss. Anyway, I then decided to make my own fork, with all the most important (IMO) patches from various sources and compiler optimizations shared by Rob. Thorium is my favourite element, and thus Thorium was born. I daily drive it and always test my own releases before uploading. Rob shared Jerry's (chromium.woolyss.com creator) contact info, and I asked him and showed him my project, and he accepted! It is a great honour and very exciting to be part of chromium.woolyss, and this is one of the reasons - gaining more users!

As far as helping, I'd love testers for various platforms (I build for windows, other architectures, and just recently Android). And also I would like someone to help build, as I'm on an overclocked AMD FX-8370 and even at that it takes 3-4 hours to compile. If you or someone else has a more powerful system It would help. The readme has instructions on building but if you or anyone else wanted to help build, I would give an interactive demo on how to build it sucessfully. I am down to video chat, give out my email, and have just about every social media besides tiktok (gross lol) It would also allow you or someone else to test their own changes, or tailor it to their system (by setting AVX2 or AVX-512 for example, which my system lacks and I can't cross build for that, but I am getting a Haswell i7 4790K and Z97 board for christmas which will help). And my fine sir, always feel free to ask questions, offer suggestions, and make bug reports/issues. Nothing is off limits, even stuff that has nothing to do with Thorium, because I like learning and love Linux/FOSS.

Again thank you for using Thorium, and welcome to my GitHub page(s).
I also build ChromiumOS for general pc usage, it has codecs, firmware, extra packages, and API Keys to sign in. > https://github.com/Alex313031/ChromiumOS and a pretty shell script called TrImLy to automate defragging/trimming depending on HDD/SSD for ChromiumOS, for which it doesn't do automatically like ChromeOS does on chromebooks > https://github.com/Alex313031/TrImLy
And I also build my own chromium app, called Chromium Dev Editor, which is a revived version of an abandoned project by google to make an IDE for Chrome written in the DART language. > https://github.com/Alex313031/Chromium-Dev-Editor

Also (sorry this is getting long) I have various snippets and small .bat/.bash/.vbs scripts one might find useful here > https://github.com/Alex313031/Snippets and here > https://github.com/Alex313031/PageFileMon

Last thing. Tell me about yourself. Age/Country/State/Interests/Job. How you became a Chromium user/lover. And YOUR projects you'd like to share with me (I'm gonna look through your GitHub right now but share anything you wish)

Cheers, m8 - Alex313031

**Comment by midzer** [comment]
Thanks for your introduction :)

Probably I can't help out with building Thorium as I'm running a rather ancient machine (i7-2600k Sandy Bridge) myself, too. Thats why I'm highly interested in those Thorium-Special AVX only builds. I made another issue and requested ARM32 builds for my Samsung S5 (from 2014).

If the project evolves and gets bigger, we might need a proper website for it. GitHub Pages will do at the beginning. Later we can get thorium-browser.com or similar. I'm familiar with web dev, so I can help there.

**Comment by midzer** [comment]
I don't know how much FMA4 is worth it, but we could use it for Thorium repo and not remove it.

We could use Thorium-Special for ARM32/64 and older CPU builds (SSE4 only and AVX only).

Yeah, I OC'd my CPU yesterday to 4,5GHz in Turbo Mode Single Core :D

Jup, av1me uses ffmpeg and is a batch tool for many videos in a folder to convert them.

I'm from Munich, Bavaria as stated in my profile. Check out my personal homepage for more info (some german there though).

**Comment by midzer** [comment]
Okay, did not want to interfere with build strategy, just my two cents as I'm newbie :)

Have a look at https://docs.github.com/en/pages, custom URL would be probably better if projects matures.

THX for AVX build!

I can try ChromiumOS on one of my USB sticks, yes :)

**Comment by Alex313031** [comment]
No, you're not interfering at all.
And thanks.
And you is welcomz

Are you familiar with ChromiumOS? (just the equivalent open source project to ChromeOS i.e. Chromium <> Chrome & ChromiumOS <> ChromeOS.) The main thing is that ChromiumOS can run on non chromebook hardware, although there is a project called Brunch https://github.com/sebanc/brunch that allows booting ChromeOS on many regular PC's

It can be written to USB with dd, or https://www.balena.io/etcher/ and booted like any other linux os.
Latest release is here (.7z contains the chromiumos_image.bin file) > https://github.com/Alex313031/ChromiumOS/releases/tag/R98.0.4729.0
You'll need a 16gb usb drive or bigger. Normal ChromiumOS images are 6.8 gb and can fit on a 8gb drive, but mine has added packages and firmware that bring the size to 9.8gb.
Also it has API Keys to sign in, but you'll still need to add your google account to https://groups.google.com/u/1/a/chromium.org/g/google-browser-signin-testaccounts as per https://www.chromium.org/developers/how-tos/api-keys as stated in the readme.

**Comment by Alex313031** [comment]
@midzer Going to close this issue now.
